linchpin/__init__.py

```diff
@@ -584,6 +585,13 @@ def do_action(self, provision_data, action='up', run_id=None, tx_id=None):
 
 
         return_code = 99
+        if vault_pass == '':
+            self.set_evar('vault_pass',
+                          self.ctx.get_cfg('evars',
+                                           'vault_pass',
+                                           default=''))
+        else:
+            self.set_evar('vault_pass', vault_pass)
```

Comment:
`do_action` should take whatever the value is in `get_evar` since it would be set as described above. Also, no need to use `get_cfg` here, `get_evar` exists to pull evar values more easily.